linux_docker_resources/Dockerfile

```diff
@@ -122,6 +122,15 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
   uncrustify \
   yamllint
 
+# Configure proposed source, and install python3-flake8 from it
+RUN if test \( ${UBUNTU_DISTRO} = noble \); then\
+    printf "deb http://archive.ubuntu.com/ubuntu/ "$(lsb_release -cs)"-proposed restricted main multiverse universe\n" >> /etc/apt/sources.list.d/ubuntu-$(lsb_release -cs)-proposed.list && \
+    printf "Package: *\nPin: release a="$(lsb_release -cs)"-proposed\nPin-Priority: -10\n" >> /etc/apt/preferences.d/proposed-updates && \
+    apt-get update &&  \
+    apt-get install --no-install-recommends -y python3-flake8/$(lsb_release -cs)-proposed \
```

Comment:
I'm still of the opinion that we should remove both the `/etc/apt/sources.list.d/ubuntu-noble-proposed.list` and `/etc/apt/preferences.d/proposed-updates` files once we are done installing.  That way, we won't be pulling metadata updates for the `noble-proposed` for all other `apt` calls below.

That said, I'm not going to block on this.  If you think we don't need to remove them, then we can leave it as-is.

Reply:
I went ahead and made this change.  I'm going to run quick CI on this, then go ahead and merge.